crates/noirc_evaluator/src/ssa.rs

```diff
@@ -104,6 +105,9 @@ pub fn create_circuit(
         public_parameters,
         return_values,
     };
+
+    let locations = locations.into_iter().map(|(k, v)| (k, v.into_iter().collect())).collect();
```

Comment:
nit: perhaps use words that specify what the key and value is instead of `k` and `v`